airflow/providers/opensearch/operators/opensearch.py

```diff
@@ -0,0 +1,192 @@
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:OpenSearchCreateIndexOperator`
+
+    :param: index_name: The name of the index to be created.
+    :param: index_body: A dictionary that defines index settings
+    :param: opensearch_conn_id: opensearch connection to use
+    """
+
+    def __init__(
+        self,
+        *,
+        index_name: str,
+        index_body: dict[str, Any],
+        opensearch_conn_id: str = "opensearch_default",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.index_name = index_name
+        self.index_body = index_body
+        self.opensearch_conn_id = opensearch_conn_id
+
+    @cached_property
+    def hook(self) -> OpenSearchHook:
+        """Gets an instance of an OpenSearchHook."""
+        return OpenSearchHook(open_search_conn_id=self.opensearch_conn_id, log_query=False)
+
+    def execute(self, context: Context) -> Any:
+        """Creates an index on an Open Search cluster."""
+        try:
+            self.hook.get_client.indices.create(index=self.index_name, body=self.index_body)
+        except Exception as e:
+            raise AirflowException(e)
+
+
+class OpenSearchAddDocumentOperator(BaseOperator):
+    """
+    Adds a new document to a given Index. It will either add or overwrite an existing document.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:OpenSearchAddDocumentOperator`
+
+    :param: index_name: The name of the index to put the document.
+    :param: document: A dictionary representation of the document.
+    :param: document_id: The id for the document in the index.
+    :param: doc_class: A Document subclassed object using opensearch-dsl
+    :param: opensearch_conn_id: opensearch connection to use
+    """
+
+    def __init__(
+        self,
+        *,
+        index_name: str | None = None,
+        document: dict[str, Any] | None = None,
+        doc_id: int | None = None,
+        doc_class: Any | None = None,
+        opensearch_conn_id: str = "opensearch_default",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.index_name = index_name
+        self.document = document
+        self.doc_id = doc_id
+        self.doc_class = doc_class
+        self.opensearch_conn_id = opensearch_conn_id
+
+    @cached_property
+    def hook(self) -> OpenSearchHook:
+        """Gets an instance of an OpenSearchHook."""
+        return OpenSearchHook(open_search_conn_id=self.opensearch_conn_id, log_query=False)
+
+    def execute(self, context: Context) -> Any:
+        """Saves a document to a given index on an OpenSearch cluster."""
+        if self.doc_class is not None:
+            try:
+                doc = self.doc_class.init(using=self.hook.get_client)
+                result = doc.save(using=self.hook.get_client)
+            except Exception as e:
```

Comment:
[Here and below] I'd prefer to see a more narrow catch here.  Catching `Exception` is generally considered poor form, in my experience.

Reply:
Fixed to use the base  OpenSearchException for the OpenSearchClient instead. It would get very verbose otherwise to go to individual exceptions as there are a potential of 15 different exceptions possible.